src/main/cpp/pop/PopulationModel.h

```diff
@@ -0,0 +1,115 @@
+#ifndef POPULATION_MODEL_H_INCLUDED
+#define POPULATION_MODEL_H_INCLUDED
+
+/**
+ * @file
+ * Header file for the population model class
+ */
+
+#include <vector>
+#include <boost/property_tree/ptree.hpp>
+
+namespace stride {
+
+/**
+ * Data from which populations may be generated.
+ */
+struct PopulationModel
+{
+	PopulationModel(const boost::property_tree::ptree& pt);
```

Comment:
I'm sort of in two minds about this constructor. I get that parsing `PopulationModel` instances straight from `ptree`s makes sure that the `PopulationModel` instances are more or less well-formed. But what I don't like about this constructor is that it
  * does work (heck, it might ever throw an exception, right?) and,
  * it makes it hard to build a `PopulationModel` instance programmatically.

I recommend replacing this constructor by a `static` method called `parse` or something.